src/test/javascript/spec/component/course/course-exercises.component.spec.ts

```diff
@@ -103,28 +96,28 @@ describe('CourseExercisesComponent', () => {
     });
 
     afterEach(() => {
-        sinon.restore();
-        jest.clearAllMocks();
+        jest.restoreAllMocks();
     });
 
     it('should initialize', () => {
-        expect(component.course).to.deep.equal(course);
-        expect(courseCalculationSpy.mock.calls.length).to.equal(2);
-        expect(courseCalculationSpy.mock.calls[0][0]).to.eq(course.id);
-        expect(courseCalculationSpy.mock.calls[1][0]).to.eq(course.id);
+        expect(component.course).toEqual(course);
+        expect(courseCalculationSpy.mock.calls.length).toEqual(2);
```

Comment:
Either use `toBe(2)` or if possible check the length with `toHaveLength(2)`
```suggestion
        expect(courseCalculationSpy.mock.calls).toHaveLength(2);
```